submissions/balancer-design/universal_8to8.md

```diff
@@ -0,0 +1,88 @@
+
+* Input-balanced: Draws from all inputs evenly. Especially important when merging many inputs to a few outputs.
+* Output-balanced: Gives to all outputs evenly. Especially important when splitting a few inputs to many outputs.
+* Throughput-limited: Has internal bottlenecks. An example of a throughput-limited system would be merging two belts into one, and then splitting it back into two. Such a system has a maximum throughput of one belt, even though two belts are available both on input and output.
+* Throughput-unlimited: Opposite of throughput-limited.
+
+All blueprints from this article (and a couple extras) can be found at [this link](https://controlc.com/70909aae).
+
+### The premise
+
+Picture the following: you're a new player, and you've just heard about those things called "balancers" and how useful they are. You see an image of a 4-4 balancer, build it, but then only hook up 3 outputs to get 3 balanced belts. Instead you're met with a 2:1:1 ratio and disappointment:
+
+![How not to use a balancer](failed_1-3_compressed.jpg)
+
+In this article, I set myself a goal to make a balancer for which this sort of intuition works, in other words, no matter the input/output configuration, all outputs are always equal, and all inputs are drawn from evenly. This kind of balancer is known as a "universal balancer", a concept first explored and named in [this post](https://www.reddit.com/r/factorio/comments/a5ferf/i_present_to_you_the_44_universal_balancer/) by Reddit user [u/tzwaan](https://www.reddit.com/user/tzwaan), who is one of the moderators of the [Factorio subreddit](https://www.reddit.com/r/factorio).
+
+Universal balancers can do what no balancer should be able to: no matter the configuration, they're input-balanced, output-balanced, *and* throughput-unlimited at the same time! This is a set of properties previously thought to be possessed only by a 2-2 balancer, which is just a single splitter. The drawback is that they get *big*.
+
+### Design principles
+
+Let's think about how to convert a 4-4 balancer into a 3-3. That is done by looping the extra output back into the first splitter. We will use this idea as our base principle:
+
+![Looped 4-4](loop_example_compressed.jpg)
+
+Now, how would we make it so the balancer automatically does this when we block some outputs? Simple, we use priority splitters to auto-loop the items that can't get out. Of course, we need to merge them back with the input too; we use input priority splitters for this, so the main inputs don't slow down when there are items coming. Here is how that might look like:
```

Comment:
> Now, how would we make it so the balancer automatically does this when we block some outputs?

"This will help that the balancer doesn't bottleneck, but instead recycle blocked items to maintain throughput"

Why would you want to do that? I know it _seems_ obvious, but it's important to make sure the reader follows.